pkg/controller/pravegacluster/upgrade.go

```diff
@@ -111,10 +121,61 @@ func (r *ReconcilePravegaCluster) clearUpgradeStatus(p *pravegav1alpha1.PravegaC
 	return nil
 }
 
-func (r *ReconcilePravegaCluster) syncComponentsVersion(p *pravegav1alpha1.PravegaCluster) (err error) {
-	var synced bool
+func (r *ReconcilePravegaCluster) rollbackClusterVersion(p *pravegav1alpha1.PravegaCluster, version string) (err error) {
+	_, rollbackCondition := p.Status.GetClusterCondition(pravegav1alpha1.ClusterConditionRollback)
+	if rollbackCondition == nil || rollbackCondition.Status != corev1.ConditionTrue {
+		// We're in the first iteration for Rollback
+		// Add Rollback Condition to Cluster Status
+		log.Printf("Updating Target Version to  %v", version)
+		p.Status.TargetVersion = version
+		p.Status.SetRollbackConditionTrue("", "")
+		updateErr := r.client.Status().Update(context.TODO(), p)
```

Comment:
What's `context.TODO()`?

Reply:
context.TODO() creates an empty context.
https://blog.golang.org/context

Reply:
The documentation says:

> TODO returns a non-nil, empty Context. Code should use context.TODO when it's unclear which Context to use or it is not yet available (because the surrounding function has not yet been extended to accept a Context parameter).

https://golang.org/pkg/context/#TODO

Could you justify why you are using it a good number of times in this PR?

Reply:
All operator code I have seen uses "context.TODO()" in every client call. 
While it does seem to work, I agree it could be suboptimal. Need to spend some time understanding this, its impact and alternatives.

Reply:
could create an issue and reference it here, please?

Reply:
https://github.com/pravega/pravega-operator/issues/266